pandas/tests/frame/test_dtypes.py

```diff
@@ -1063,6 +1063,17 @@ def test_asarray_homogenous(self):
         expected = np.array([[1, 1], [2, 2]], dtype="object")
         tm.assert_numpy_array_equal(result, expected)
 
+    def test_str_to_small_float_conversion_type(self):
+        # GH 20388
+        np.random.seed(13)
+        col_data = [str(np.random.random() * 1e-12) for _ in range(5)]
+
+        df = pd.DataFrame(col_data, columns=["A"])
+        assert df["A"].dtype == np.object_
```

Comment:
For these two assertions, can you construct the expected dataframe result and use `tm.assert_frame_equal`?

```
result = ...
expected = ...
tm.assert_frame_equal(result, expected)
```

Reply:
Thanks for the feedback, @mroeschke! Changed the test accordingly.